src/Config.php

```diff
@@ -8,18 +8,34 @@
 
 class Config extends BaseConfig
 {
-    public function getToken() : string
+    public const MODE_UPDATE = 'update';
+    public const MODE_REPLACE = 'replace';
+    public const MODE_RECREATE = 'recreate';
+
+    public function getToken(): string
     {
-        return $this->getValue(['parameters', '#token']);
+        return (string) $this->getValue(['parameters', '#token']);
     }
 
-    public function getUrl() : string
+    public function getUrl(): string
     {
-        return $this->getValue(['parameters', 'url']);
+        return (string) $this->getValue(['parameters', 'url']);
     }
 
-    public function isIncremental() : bool
+    public function getMode(): string
     {
-        return $this->getValue(['parameters', 'incremental']);
+        // can't pass null as default value, because it represents no default.
+        // Abuse 0 to identify that incremental flag is not set at all.
+        $incremental = $this->getValue(['parameters', 'incremental'], 0);
+        // the legacy incremental flag is replaced by the `mode` parameter
+        if ($incremental === 0) {
```

Comment:
tohle je trochu blbý v tom, že když nastavíš současně `incremental` a `mode`, tak incremental má přednost, i když bys řekl, že už je deprecated. ale nevím, jak jinak to vyřešit